SYCL/Basic/barrier_order.cpp

```diff
@@ -11,28 +10,57 @@ int main() {
   sycl::device dev{sycl::default_selector{}};
   sycl::queue q{dev};
 
-  int *x = sycl::malloc_shared<int>(1, q);
-  int *y = sycl::malloc_shared<int>(1, q);
+  unsigned long long *x = sycl::malloc_shared<unsigned long long>(1, q);
+
+  int error = 0;
+
+  // test barrier without arguments
   *x = 0;
-  *y = 0;
+  for (int i = 0; i < 64; i++) {
+    q.single_task([=]() {
+      // do some busywork
+      float y = *x;
```

Comment:
```suggestion
      volatile float y = *x;
```
Otherwise `y` will be considered dead and the busywork will be removed by the optimizer.

Reply:
Makes sense, although the test worked even without that.

Reply:
Could have had enough work on the platform you tested on. I suppose the work isn't guaranteed to actually cause it to be running when the barriers are submitted, but the hope is to increase the chance of it. 😄

Reply:
No, the presence of the loop is what made the difference between passing or failing, soI guess even without volatile it was for some reason not optimized away.

Reply:
I checked and the cos operation disappears, but I can confirm that there is still a loop in the produced IR. That said, since the loop is empty there's a chance it could be optimized out in the future or that a higher optimization level would have removed it, so it is definitely safer this way.